v03_pipeline/lib/model/reference_dataset_collection.py

```diff
@@ -94,3 +94,40 @@ def for_reference_genome_dataset_type(
         if not Env.ACCESS_PRIVATE_REFERENCE_DATASETS:
             return [rdc for rdc in rdcs if rdc.access_control == AccessControl.PUBLIC]
         return rdcs
+
+    @classmethod
+    def for_dataset(cls, dataset: str) -> 'ReferenceDatasetCollection':
```

Comment:
It'd be cool if we could use the existing method so we don't have to duplicate lists.  It's basically just calling `datasets(dataset_type)` for all possible`DatasetType`?

Reply:
ok I cleaned this up to use `datasets` function, and I added `dataset_type` as an arg to `for_dataset` since we have it from the calling task.